File: Advance/14.Write a program to compute the nth Catalan number/Solution.cpp
#include <iostream>
using namespace std;

unsigned long long factorial(int n) {
    unsigned long long result = 1;
    for (int i = 2; i <= n; i++) {
        result *= i;
    }
    return result;
}

unsigned long long catalanNumber(int n) {
    unsigned long long numerator = factorial(2 * n);
    unsigned long long denominator = factorial(n + 1) * factorial(n);
    return numerator / denominator;
}

int main() {
    int n;
    cout << "Enter the value of n: ";
    cin >> n;

    cout << "The " << n << "th Catalan number is: " << catalanNumber(n) << endl;
    return 0;
}

File: Advance/14.Write a program to compute the nth Catalan number/Solution.java
import java.util.Scanner;

public class CatalanNumber {

    public static long factorial(int n) {
        long result = 1;
        for (int i = 2; i <= n; i++) {
            result *= i;
        }
        return result;
    }

    public static long catalanNumber(int n) {
        long numerator = factorial(2 * n);
        long denominator = factorial(n + 1) * factorial(n);
        return numerator / denominator;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of n: ");
        int n = sc.nextInt();

        System.out.println("The " + n + "th Catalan number is: " + catalanNumber(n));
    }
}
